app/src/test/java/com/duckduckgo/app/settings/SettingsViewModelTest.kt

```diff
@@ -111,9 +110,8 @@ class SettingsViewModelTest {
     fun before() {
         MockitoAnnotations.openMocks(this)
 
-        whenever(networkProtectionWaitlist.getState()).thenReturn(NetPWaitlistState.NotUnlocked)
```

Comment:
@aitorvs also looking into[ SettingsViewmodel](https://github.com/duckduckgo/Android/blob/8c106c192a42d84c8ab7ae67057a5b899afac95a/app/src/main/java/com/duckduckgo/app/settings/SettingsViewModel.kt#L206), `getNetworkProtectionEntryState` which calls `getState()` i s being executed in the main thread. Either we change the flowOn to io or move   networkProtectionWaitlist.getState() somewhere else?

Reply:
Resolved. We `getState` in `io` so the piece of logic in question is no longer relevant.